tests/dhcp_server/test_dhcp_server_stress.py

```diff
@@ -0,0 +1,198 @@
+    }
+
+    logging.info("The vlan_setting before test is %s" % vlan_setting)
+    return vlan_setting['vlan_name'], \
+        vlan_setting['vlan_gateway'], \
+        vlan_setting['vlan_subnet_mask'], \
+        vlan_setting['vlan_hosts'], \
+        vlan_setting['vlan_member_with_ptf_idx']
+
+
+def test_dhcp_server_with_multiple_dhcp_clients(
+    duthost,
+    ptfhost,
+    parse_vlan_setting_from_running_config
+):
+    """
+        Make sure all ports can get assigend ip when all ports request ip at same time
+    """
+    vlan_name, gateway, net_mask, vlan_hosts, vlan_members_with_ptf_idx = parse_vlan_setting_from_running_config
+    start_command = " && ".join(["dhclient -nw eth%s" % ptf_index for _, ptf_index in vlan_members_with_ptf_idx])
+    end_command = " ; ".join(["dhclient -r eth%s" % ptf_index for _, ptf_index in vlan_members_with_ptf_idx])
+    try:
+        config_to_apply = empty_config_patch()
+        dut_ports, _ = zip(*vlan_members_with_ptf_idx)
+        exp_assigned_ip_ranges = [[ip] for ip in vlan_hosts[:len(vlan_members_with_ptf_idx)]]
+        append_common_config_patch(
+            config_to_apply,
+            vlan_name,
+            gateway,
+            net_mask,
+            dut_ports,
+            exp_assigned_ip_ranges
+        )
+        apply_dhcp_server_config_gcu(duthost, config_to_apply)
+        ptfhost.shell(start_command)
+
+        def all_ip_shown_up(ptfhost, expected_assigned_ips):
+            ip_addr_output = ptfhost.shell("ip addr")['stdout']
+            for expected_ip in expected_assigned_ips:
+                if expected_ip not in ip_addr_output:
+                    return False
+            return True
+        expected_assigned_ips = [range[0] for range in exp_assigned_ip_ranges]
+        pytest_assert(
+            wait_until(20, 1, 1,
+                       all_ip_shown_up,
+                       ptfhost,
+                       expected_assigned_ips),
+            'Not all configurated IP shown up on ptf interfaces'
+        )
+    finally:
+        ptfhost.shell(end_command, module_ignore_errors=True)
+        ptfhost.shell("killall dhclient", module_ignore_errors=True)
+
+
+@pytest.mark.parametrize("stress_config", [(20, 1), (40, 1), (80, 2), (100, 7), (200, 8), (400, 10), (700, 10)])
+def test_dhcp_server_with_large_number_of_discover(
+    duthost,
+    ptfhost,
+    ptfadapter,
+    stress_config,
+    parse_vlan_setting_from_running_config
+):
+    config_to_apply = empty_config_patch()
+    vlan_name, gateway, net_mask, vlan_hosts, vlan_members_with_ptf_idx = parse_vlan_setting_from_running_config
+    exp_assigned_ip_ranges = [[ip] for ip in vlan_hosts[:len(vlan_members_with_ptf_idx)]]
+    dut_ports, ptf_port_indexs = zip(*vlan_members_with_ptf_idx)
+    logging.info("expected_assigned_ip_rangs is %s, dut_ports is %s, ptf_port_indexs is %s" %
+                 (exp_assigned_ip_ranges, dut_ports, ptf_port_indexs))
+    append_common_config_patch(
+        config_to_apply,
```

Comment:
Looks like the dhcp_server configuration is always the same for different packet rate? Then would it be better to apply this configuration in a module-level fixture, then it will only be involved 1 time

Reply:
And if test_dhcp_server_with_multiple_dhcp_clients and test_dhcp_server_with_large_number_of_discover share the same configuration, we can reuse the fixture too

Reply:
I prefer everytime they can setup new configuration.
We only change dhcp_server configuration, it's lightweight.